.github/workflows/ci-tests.yml

```diff
@@ -36,13 +42,12 @@ jobs:
     - name: Install requirements
       run: |
         sudo apt-get update
-        sudo apt-get install firefox firefox-geckodriver gettext gettext-base xvfb
         pip install tox
+    - name: Seting URL environment variable, somehow localhost not working.
+      run: |
+        echo "URL=http://`hostname -i`:8522" >> $GITHUB_ENV
```

Comment:
WAT? When I saw you cast this magic spell, I did not understand it. So I went here:

https://explainshell.com/explain?cmd=echo+%22URL%3Dhttp%3A%2F%2F%60hostname+-i%60%3A8522%22+%3E%3E+%24GITHUB_ENV

And I still don't understand it. I just know that without it, we do not set `URL` in the environment. Is there a cleaner way to set this env var?

Reply:
This is to set URL based on local ip address as we explained it in [document](https://github.com/Pylons/deform/blob/master/contributing.md#url)

The default URL not working with localhost or 127.0.0.1, either in local tests using container or running container in Github Workflows, so we should inject VM ip address to URL so the browser that started within container connects to pserve that started within VM. I think browser would think localhost is the container OS local host, and that is the reason it fails. 

This is the only way I found to run a shell command to get VM ip address dynamically instead of type it staticly.

Reply:
$GITHUB_ENV is to set and share environment variable with other steps within the same job.
Explained [here](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable)

Reply:
We need to get the ip address of local VM and inject it to URL, unless type the VM ip address statically which might change eventually.

A cleaner way maybe, remove the back tick and replace it with paranthesis:
```
echo "URL=http://$(hostname -i):8522" >> $GITHUB_ENV
```

Either is shoudl be ip address of hostname -i to get the VM ip address.

Reply:
OK, thanks for the link to the GHA docs. That was the nugget of information that I didn't have.